### DISCUSSION TITLE
Why does imaginary frequency appears just when the gradient of free energy reaches 0?

### DISCUSSION THREAD

**Discussion by biglinn** [SSCHAcode/python-sscha#351] [Q&A]
Hi, @mesonepigreco:
I find a strange situation. I have used dynnamical matrices with 3x3x3 q-mesh to get 3x3x3 supercell.

When SSCHA goes to the 9th iteration, it shows good phonon dispersion. But the gradient has not yet reached 0, as follows:

![population9](https://github.com/user-attachments/assets/9b1919b8-3150-42f6-b6ac-443360d45f71)
![image](https://github.com/user-attachments/assets/f781b6f9-1346-4da6-83bc-5c7cab0a64f3)

When SSCHA goes to the 10th iteration, imaginary frequency appears. However the gradient reaches 0. And subsequent iterations perform like the 10th.
The 10th iteration is as follows:

![population10](https://github.com/user-attachments/assets/fa075767-d7e1-4754-9a91-17a7a176848b)
![image](https://github.com/user-attachments/assets/161cc30e-9532-4985-b270-3a6a6f0e6c52)

Does it mean that the 10th iteration with imaginary frequency should be the result of convergence?

**Comment by mesonepigreco** [comment]
Hi, the imaginary frequency you see in the last converged iteration is in an interpolated point, so it could be just an artifact of the interpolation. 

You can try to interpolate the difference of your last SSCHA dynamical matrix with the starting one on a harmonic calculation performed on a finer mesh to have a better phonon dispersion in points that are noncommensurate with your 3x3x3 supercell.

**Reply by biglinn** [reply]
Sorry, I don't quite understand what you said 
> interpolate the difference of your last SSCHA dynamical matrix with the starting one

I don't use `Interpolate` function for the above results, but use QE to calculate harmonic dyns with 3x3x3 q-mesh in order to get 3x3x3 supercell.
